hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/AbstractAssociatePrecompile.java

```diff
@@ -0,0 +1,112 @@
+package com.hedera.services.store.contracts.precompile.impl;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.context.SideEffectsTracker;
+import com.hedera.services.context.primitives.StateView;
+import com.hedera.services.contracts.sources.EvmSigsVerifier;
+import com.hedera.services.fees.FeeCalculator;
+import com.hedera.services.ledger.accounts.ContractAliases;
+import com.hedera.services.store.contracts.WorldLedgers;
+import com.hedera.services.store.contracts.precompile.InfrastructureFactory;
+import com.hedera.services.store.contracts.precompile.Precompile;
+import com.hedera.services.store.contracts.precompile.SyntheticTxnFactory;
+import com.hedera.services.store.contracts.precompile.codec.Association;
+import com.hedera.services.store.contracts.precompile.codec.DecodingFacade;
+import com.hedera.services.store.contracts.precompile.utils.KeyActivationUtils;
+import com.hedera.services.store.contracts.precompile.utils.PrecompilePricingUtils;
+import com.hedera.services.store.models.Id;
+import com.hederahashgraph.api.proto.java.Timestamp;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+
+import javax.inject.Provider;
+import java.util.Objects;
+
+import static com.hedera.services.exceptions.ValidationUtils.validateTrue;
+import static com.hedera.services.store.contracts.precompile.utils.PrecompilePricingUtils.GasCostType.ASSOCIATE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_SIGNATURE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+
+/* --- Constructor functional interfaces for mocking --- */
+public abstract class AbstractAssociatePrecompile implements Precompile {
```

Comment:
This class can extend `AbstractWritePrecompile` class. In this way a lot of the fields would be removed from here.